app/models/bulkrax/csv_entry.rb

```diff
@@ -149,7 +151,7 @@ def build_mapping_metadata
       mapping = fetch_field_mapping
       mapping.each do |key, value|
         # these keys are handled by other methods
-        next if ['model', 'file', related_parents_parsed_mapping, related_children_parsed_mapping].include?(key)
+        next if ['model', 'file', related_parents_parsed_mapping, related_children_parsed_mapping, source_identifier].include?(key)
```

Comment:
we're already adding the `source_identifier` to `self.parsed_metadata` in `#build_system_metadata`